### ISSUE TITLE
Ability to sync SOME files out of a container to the host

### ISSUE THREAD

**Issue by djcp** [windmilleng/tilt#3155]
**tl;dr**

There are some legit cases where it would be natural to sync changes OUT of a container and into the host, especially around dependency management. The alternative - having to manage essentially two copies of the language environment in the host and containers - obviates some of why you'd use docker / containers for dev.

<hr></hr>

Please see: https://kubernetes.slack.com/archives/CESBL84MV/p1585675286052500 (which is summarized below):

Let's say there's a process that needs to run in a tilt-managed pod that modifies a file that should be checked into git. For example, with ruby (and js and other languages), if I want to include a new dependency, I:

* edit `Gemfile` to include the dependency
* run bundle, which writes out a `Gemfile.lock` 
* I then check `Gemfile` and `Gemfile.lock` into git, and all instances of this app have the same dependencies.

The problem however, is that `Gemfile.lock` is best run in the container, as it has the ruby environment and will be able to resolve dependencies correctly.

So it seems we have three options:

1. Require that devs have the requisite version of ruby and all its dependencies in the host, so they can bundle in two environments: `tilt`, and locally to be able to check the change to `Gemfile.lock` into git
2. Figure out a way to get changes out of a tilt managed pod
3. use `kubectl cp`

Option 1 kinda sucks, as it means we still have to worry about setting up ruby in the host and obviates some of the beauty of docker/k8s

Option 2: Am I correct in thinking tilt syncing is only unidirectional from the host to the container? It seems that way from reading docs and experimentation. What are my workarounds?

Option 3 would work but something more automatic and natural would be very welcome

**Comment by djcp** [comment]
I guess we could use `kubectl cp` as a last resort, but something more robust and automatic would be ideal

**Comment by djcp** [comment]
This also means any code that generates code (a database migrations, other kinds of templatized code generation, etc) would have to be run only in the host as well.

**Comment by landism** [comment]
Nice writeup! I think this is the same as #1956.

Your assessment of the situation is pretty much accurate.

AFAIK the best current workaround is basically `kubectl cp`, as you suggest, though it could be a little nicer by wrapping it in a tilt [`local_resource`](https://docs.tilt.dev/api.html#api.local_resource) w/ [`trigger_mode=TRIGGER_MODE_MANUAL`](https://docs.tilt.dev/manual_update_control.html) so that users can just click a button in the Tilt UI to sync the gems back to their disk.

Feel free to reopen this if I've misunderstood, or comment over in #1956.
